AccessAnalyzer sweepers

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Description

<!--- Please leave a helpful description of the feature request here. --->

```
=== RUN   TestAccAccessAnalyzer_serial
=== RUN   TestAccAccessAnalyzer_serial/Analyzer
=== RUN   TestAccAccessAnalyzer_serial/Analyzer/basic
analyzer_test.go:24: Step 1/2 error: Error running apply: exit status 1
Error: error creating Access Analyzer Analyzer (tf-acc-test-3567209128705452227): ServiceQuotaExceededException: Analyzer limit exceeded
{
RespMetadata: {
StatusCode: 402,
RequestID: "986c12eb-81a5-4704-ab0b-7220e454ff59"
},
Message_: "Analyzer limit exceeded"
}
with aws_accessanalyzer_analyzer.test,
on terraform_plugin_test.tf line 2, in resource "aws_accessanalyzer_analyzer" "test":
2: resource "aws_accessanalyzer_analyzer" "test" {
=== RUN   TestAccAccessAnalyzer_serial/Analyzer/disappears
analyzer_test.go:56: Step 1/1 error: Error running apply: exit status 1
Error: error creating Access Analyzer Analyzer (tf-acc-test-7228912598435599031): ServiceQuotaExceededException: Analyzer limit exceeded
{
RespMetadata: {
StatusCode: 402,
RequestID: "ddf9d11f-0831-4923-8254-e5fc8f3d5c71"
},
Message_: "Analyzer limit exceeded"
}
with aws_accessanalyzer_analyzer.test,
on terraform_plugin_test.tf line 2, in resource "aws_accessanalyzer_analyzer" "test":
2: resource "aws_accessanalyzer_analyzer" "test" {
=== RUN   TestAccAccessAnalyzer_serial/Analyzer/Tags
analyzer_test.go:81: Step 1/4 error: Error running apply: exit status 1
Error: error creating Access Analyzer Analyzer (tf-acc-test-1196313018691509922): ServiceQuotaExceededException: Analyzer limit exceeded
{
RespMetadata: {
StatusCode: 402,
RequestID: "ee1e733c-1cb9-45c3-8b8f-7688a3541c52"
},
Message_: "Analyzer limit exceeded"
}
with aws_accessanalyzer_analyzer.test,
on terraform_plugin_test.tf line 2, in resource "aws_accessanalyzer_analyzer" "test":
2: resource "aws_accessanalyzer_analyzer" "test" {
=== RUN   TestAccAccessAnalyzer_serial/Analyzer/Type_Organization
acctest.go:652: skipping tests; this AWS account must not be an existing member of an AWS Organization
--- FAIL: TestAccAccessAnalyzer_serial (19.43s)
--- FAIL: TestAccAccessAnalyzer_serial/Analyzer (19.43s)
--- FAIL: TestAccAccessAnalyzer_serial/Analyzer/basic (7.42s)
--- FAIL: TestAccAccessAnalyzer_serial/Analyzer/disappears (5.60s)
--- FAIL: TestAccAccessAnalyzer_serial/Analyzer/Tags (6.03s)
--- SKIP: TestAccAccessAnalyzer_serial/Analyzer/Type_Organization (0.37s)
```

--- a/internal/service/accessanalyzer/analyzer.go
+++ b/internal/service/accessanalyzer/analyzer.go
@@ -169,19 +169,18 @@ func resourceAnalyzerUpdate(d *schema.ResourceData, meta interface{}) error {
 func resourceAnalyzerDelete(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*conns.AWSClient).AccessAnalyzerConn
 
-	input := &accessanalyzer.DeleteAnalyzerInput{
+	log.Printf("[DEBUG] Deleting Access Analyzer Analyzer: (%s)", d.Id())
+	_, err := conn.DeleteAnalyzer(&accessanalyzer.DeleteAnalyzerInput{
 		AnalyzerName: aws.String(d.Id()),
 		ClientToken:  aws.String(resource.UniqueId()),
-	}
+	})
 
-	_, err := conn.DeleteAnalyzer(input)
-
-	if tfawserr.ErrMessageContains(err, accessanalyzer.ErrCodeResourceNotFoundException, "") {
+	if tfawserr.ErrCodeEquals(err, accessanalyzer.ErrCodeResourceNotFoundException) {
 		return nil
 	}
 
 	if err != nil {
-		return fmt.Errorf("error deleting Access Analyzer Analyzer (%s): %s", d.Id(), err)
+		return fmt.Errorf("error deleting Access Analyzer Analyzer (%s): %w", d.Id(), err)
 	}
 
 	return nil

--- a/internal/service/accessanalyzer/sweep.go
+++ b/internal/service/accessanalyzer/sweep.go
@@ -1,0 +1,65 @@
+//go:build sweep
+// +build sweep
+
+package accessanalyzer
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/accessanalyzer"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+	"github.com/hashicorp/terraform-provider-aws/internal/sweep"
+)
+
+func init() {
+	resource.AddTestSweepers("aws_accessanalyzer_analyzer", &resource.Sweeper{
+		Name: "aws_accessanalyzer_analyzer",
+		F:    sweepAnalyzers,
+	})
+}
+
+func sweepAnalyzers(region string) error {
+	client, err := sweep.SharedRegionalSweepClient(region)
+	if err != nil {
+		return fmt.Errorf("error getting client: %s", err)
+	}
+	conn := client.(*conns.AWSClient).AccessAnalyzerConn
+	input := &accessanalyzer.ListAnalyzersInput{}
+	sweepResources := make([]*sweep.SweepResource, 0)
+
+	err = conn.ListAnalyzersPages(input, func(page *accessanalyzer.ListAnalyzersOutput, lastPage bool) bool {
+		if page == nil {
+			return !lastPage
+		}
+
+		for _, analyzer := range page.Analyzers {
+			r := ResourceAnalyzer()
+			d := r.Data(nil)
+			d.SetId(aws.StringValue(analyzer.Name))
+
+			sweepResources = append(sweepResources, sweep.NewSweepResource(r, d, client))
+		}
+
+		return !lastPage
+	})
+
+	if sweep.SkipSweepError(err) {
+		log.Printf("[WARN] Skipping Access Analyzer Analyzer sweep for %s: %s", region, err)
+		return nil
+	}
+
+	if err != nil {
+		return fmt.Errorf("error listing Access Analyzer Analyzers (%s): %w", region, err)
+	}
+
+	err = sweep.SweepOrchestrator(sweepResources)
+
+	if err != nil {
+		return fmt.Errorf("error sweeping Access Analyzer Analyzers (%s): %w", region, err)
+	}
+
+	return nil
+}

--- a/internal/sweep/sweep_test.go
+++ b/internal/sweep/sweep_test.go
@@ -7,6 +7,7 @@ import (
 	"testing"
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	_ "github.com/hashicorp/terraform-provider-aws/internal/service/accessanalyzer"
 	_ "github.com/hashicorp/terraform-provider-aws/internal/service/acm"
 	_ "github.com/hashicorp/terraform-provider-aws/internal/service/acmpca"
 	_ "github.com/hashicorp/terraform-provider-aws/internal/service/amplify"

